esphome/components/waveshare_epaper/waveshare_epaper.cpp

```diff
@@ -1507,6 +1507,326 @@ void WaveshareEPaper2P9InV2R2::set_full_update_every(uint32_t full_update_every)
+        }
+      }
+      // now calculate a CRC16_checksum and compare it against the stored value.
+      // ESP_LOGD(TAG, "Calculating a CRC");
+      uint16_t segment_crc = crc16(segment, segment_size * 2, 65535U, 40961U, false, false);
+      if (full_sync) {
+        // no need to compare, we're in the first run, just place it. This is called by full refresh only
+        this->checksums_[seg_x + seg_y * seg_x_] = segment_crc;
+      } else {
+        // Partial Update, compare checksums while replacing and record the X and Y block position of the top left and
+        // bottom right corner of the changed elements. Afterwards, we can partially update only the segment that has
+        // been altered.
+        bool changed = this->checksums_[seg_x + seg_y * seg_x_] != segment_crc;
+        this->checksums_[seg_x + seg_y * seg_x_] = segment_crc;
+        if (changed && !found_change) {
+          found_change = true;
+          // We need to span the x segment, with the lowest segment found making the first segment and the highest
+          // segment found the last segment.
+          if (seg_x < this->first_segment_x_)
+            this->first_segment_x_ = seg_x;
+          if (seg_x > this->last_segment_x_)
+            this->last_segment_x_ = seg_x;
+          this->first_segment_y_ = seg_y;
+        } else if (changed) {
+          // We need to span the x segment, with the lowest segment found making the first segment and the highest
+          // segment found the last segment.
+          if (seg_x < this->first_segment_x_)
+            this->first_segment_x_ = seg_x;
+          if (seg_x > this->last_segment_x_)
+            this->last_segment_x_ = seg_x;
+          // Segment changed but we already found a change before, so set it as last segment
+          this->last_segment_y_ = seg_y;
+        } else {
+          // do nothing, segment didn't change.
+        }
+      }
+      delete[] segment;  // Delete the heap element again, this is not java... >.<
+    }
+  }
+  if (found_change) {
+    ESP_LOGD(TAG, "CRC Calculation finished. Found changes: %02d:%02d to %02d:%02d", first_segment_x_, first_segment_y_,
+             last_segment_x_, last_segment_y_);
+  } else {
+    ESP_LOGD(TAG, "CRC Calculation finished. No changes found.");
+  }
+}
+void GDEY075Z08::set_full_update_every(uint32_t full_update_every) { this->full_update_every_ = full_update_every; }
+
+void GDEY075Z08::set_num_segments_x(uint8_t value) {
+  if (this->get_width_internal() % (value * 8) != 0) {
+    ESP_LOGD(TAG,
+             "Invalid number of X Segments (%d) The display width divided by number of segments must be divisible by "
+             "8 for "
+             "proper byte boundaries. Setting num_segments_x to 20.",
+             value);  // We actually don't have to do anything, 10 is the default value :o)
+  } else {
+    this->seg_x_ = value;
+  }
```

Comment:
This should be enforced at config validation, not at runtime.
```suggestion
  this->seg_x_ = value;
```

Reply:
I actually thought about this, and didn't know how to do this. I don't have the dimensions of the display in python (since they are managed in the C++ Code), PLUS if (as I am fully expecting), these settings will also be eventually used for other displays with different resolutions, the values would probably be different.

Reply:
Most displays either have hardcoded resolutions in the `get_width_internal` or they are set by the user in config. 
Since this only applies to this display right now, we can start a dict of model -> resolution and look it up that way.